src/main/java/org/jenkins/tools/test/model/PluginRemoting.java

```diff
@@ -106,181 +93,34 @@ private String retrievePomContentFromXmlFile() throws IOException {
-                                getValueOrFail(doc, xpath, "/project/parent/version"));
-            } else {
-                LOGGER.log(Level.FINE, "No parent POM for {0}", artifactId);
-            }
-        } catch (ParserConfigurationException | SAXException | IOException e) {
-            LOGGER.log(Level.WARNING, "Failed to parse pom.xml", e);
+        Model model;
+        try (Reader r = new StringReader(pomContent)) {
+            MavenXpp3Reader mavenXpp3Reader = new MavenXpp3Reader();
+            model = mavenXpp3Reader.read(r);
+        } catch (XmlPullParserException e) {
             throw new PluginSourcesUnavailableException("Failed to parse pom.xml", e);
-        } catch (XPathExpressionException e) {
-            LOGGER.log(Level.WARNING, "Failed to retrieve SCM connection", e);
-            throw new PluginSourcesUnavailableException("Failed to retrieve SCM connection", e);
-        }
-
-        PomData pomData =
-                new PomData(artifactId, packaging, scmConnection, scmTag, parent, groupId);
-        computeScmConnection(pomData);
-        return pomData;
-    }
-
-    /**
-     * Retrieves a field value by XPath. The value must exist and be non-empty.
-     *
-     * @throws IOException parsing error
-     */
-    @NonNull
-    private static String getValueOrFail(Document doc, XPath xpath, String field)
-            throws IOException {
-        String res;
-        try {
-            res = xpath.evaluate(field + "/text()", doc);
-        } catch (XPathExpressionException e) {
-            throw new IOException("Expression failed for the field " + field, e);
-        }
-
-        if (res == null || res.isBlank()) {
-            throw new IOException("Field is either null or blank: " + field);
-        }
-        return res;
-    }
-
-    public static void computeScmConnection(PomData pomData) {
-        String transformedConnectionUrl = pomData.getConnectionUrl();
-
-        // Trimming url
-        transformedConnectionUrl = transformedConnectionUrl.trim();
-
-        // Generally, when connectionUrl is empty, is implies it is declared in a parent pom
-        // => Only possibility is to deduct github repository from artifactId (crossing fingers it
-        // is not a bizarre repository url...)
-        String oldUrl = transformedConnectionUrl;
-        if (transformedConnectionUrl.isEmpty()) {
-            transformedConnectionUrl =
-                    "scm:git:git://github.com/jenkinsci/"
-                            + pomData.artifactId.replaceAll("jenkins", "")
-                            + "-plugin.git";
-            if (!oldUrl.equals(transformedConnectionUrl)) {
-                LOGGER.log(
-                        Level.WARNING,
-                        "project.scm.connectionUrl is not present in plugin's pom .. isn't it"
-                                + " residing somewhere on a parent pom ?");
-            }
-        }
```

Comment:
If this is actually a problem the solution is to update the parent POM to read `<scm child.scm.connection.inherit.append.path="false" child.scm.developerConnection.inherit.append.path="false" child.scm.url.inherit.append.path="false">` per our archetype.